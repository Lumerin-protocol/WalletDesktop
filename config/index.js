'use strict'
require('dotenv').config()

var ethRopstenLocal = require('./ethRopstenLocal')
var etcMordenLocal = require('./etcMordenLocal')
var ethMainnet = require('./ethMainnet')
var ethRopsten = require('./ethRopsten')
var etcMorden = require('./etcMorden')

module.exports = {
  chains: {
    ethRopstenLocal,
    etcMordenLocal,
    ethMainnet,
    ethRopsten,
    etcMorden
  },
  enabledChains: process.env.REACT_APP_ENABLED_CHAINS
    ? process.env.REACT_APP_ENABLED_CHAINS.split(' ')
    : ['ethMain'],
  requiredPasswordEntropy: 72,
  explorerDebounce: 2000,
  ratesUpdateMs: 30000,
  trackingId: process.env.TRACKING_ID || null,
  sentryDsn: process.env.SENTRY_DSN || null,
  debug: process.env.DEBUG || false
}
