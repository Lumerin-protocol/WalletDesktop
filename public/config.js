'use strict'

// TODO: check better way to handle env vars for packaged version

const config = {
  sentryDsn:
    process.env.SENTRY_DSN || 'https://d9905c2eec994071935593d4085d3547@sentry.io/290706'
}

module.exports = config
