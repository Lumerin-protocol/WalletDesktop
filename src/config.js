import MetronomeContracts from 'metronome-contracts'

const addresses = MetronomeContracts.addresses[process.env.REACT_APP_ETH_CHAIN || 'main']
const defaultExplorerUrl = 'https://explorer.metronome.io'

export default {
  MET_TOKEN_ADDR: addresses.metToken.toLowerCase(),
  CONVERTER_ADDR: addresses.autonomousConverter.toLowerCase(),
  MET_EXPLORER_URL: process.env.REACT_APP_MET_EXPLORER_URL || defaultExplorerUrl,
  SENTRY_DSN: process.env.REACT_APP_SENTRY_DSN  || 'https://d9905c2eec994071935593d4085d3547@sentry.io/290706',
  ETH_DEFAULT_GAS_LIMIT: '21000',
  MET_DEFAULT_GAS_LIMIT: '250000',
  DEFAULT_GAS_PRICE: '1000000000',
  MAX_GAS_PRICE: '20000000000000000' // ~= $12 USD
}
