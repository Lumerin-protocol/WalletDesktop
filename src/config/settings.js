const config = {
  MTN_API_URL: 'http://api.mtn.bloqrock.net',
  MTN_PUBLIC_NODE_URL: 'ws://parity.bloqrock.net:8546/',
  MTN_AUCTION_ADDR: '0xd0aa441ccc3926bcc28d586043bed845d9617ce1',
  MTN_TOKEN_ADDR: '0x2d9a998fa591ef40563dc56bac835d03680f8d23',
  WS_AUCTION_STATUS_TOPIC: 'AUCTION_STATUS_TASK',
  SECURE_STORAGE_KEY: 'seed'
};

export default config;
