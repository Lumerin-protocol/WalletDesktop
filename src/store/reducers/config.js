import { handleActions } from 'redux-actions';

const initialState = {
  chain: {}
};

const reducer = handleActions(
  {
    'initial-state-received': (_, { payload }) => payload.config,
    'buyer-default-pool-received': (state, { payload }) => ({
      ...state,
      buyerDefaultPool: payload
    }),
    'ip-received': (state, { payload }) => ({
      ...state,
      ip: payload
    }),
    'set-seller-currency': (state, { payload }) => ({
      ...state,
      sellerCurrency: payload
    })
  },
  initialState
);

export default reducer;
