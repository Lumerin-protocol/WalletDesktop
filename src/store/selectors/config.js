import { createSelector } from 'reselect';

// Returns the "config" state branch
export const getConfig = state => state.config;

export const getCoinSymbol = state => state.config.chain.symbol;

export const getSymbolEth = state => state.config.chain.symbolEth;

export const getBuyerProxyPort = state => state.config.chain.proxyPort;

export const getSellerProxyPort = state => state.config.chain.proxyPort;

export const getIsAuthBypassed = state => state.config.chain.bypassAuth;

export const getIp = state => state.config.ip;

export const getPortCheckErrorLink = state =>
  state.config.chain.portCheckErrorLink;

export const getBuyerPool = state => state.config.buyerDefaultPool;

export const getRecaptchaSiteKey = state => state.config.recaptchaSiteKey;

export const getAutoAdjustPriceInterval = state =>
  state.config.autoAdjustPriceInterval;

export const getAutoAdjustContractPriceTimeout = state =>
  state.config.autoAdjustContractPriceTimeout;

export const getFaucetUrl = state => state.config.chain.faucetUrl;
export const showFaucet = state => state.config.chain.showFaucet;

export const getSellerSelectedCurrency = state => state.config.sellerCurrency;

export const getSellerDefaultCurrency = state =>
  state.config.chain.defaultSellerCurrency;
