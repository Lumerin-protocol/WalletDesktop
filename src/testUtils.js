/* eslint-disable require-path-exists/exists */
/* eslint-disable import/no-unresolved */
import { Provider as ClientProvider } from 'lumerin-wallet-ui-logic/src/hocs/clientContext';
import { Provider, createStore } from 'lumerin-wallet-ui-logic/src/store';
import { render, Simulate } from 'react-testing-library';
import { ThemeProvider } from 'styled-components';
import { MemoryRouter } from 'react-router';
import createClient from './client';
import { merge } from 'lodash';
import theme from 'lumerin-wallet-ui-logic/src/theme';
import React from 'react';

import config from '../config';

/*
 * The same render method of 'react-testing-library' but wrapped with
 * Redux store Provider and 'styled-components' ThemeProvider.
 *
 * element      : the React element you want to wrap
 * initialState : optional state to initialize the Redux store
 *
 * Returns the same object as render() but with an additional 'store'
 * property useful for dispatching actions inside your tests.
 */
export function reduxRender(element, initialState) {
  const client = createClient(config, createStore, initialState);

  const renderResult = render(
    <ClientProvider value={client}>
      <Provider store={client.store}>
        <ThemeProvider theme={theme}>
          <React.Fragment>{element}</React.Fragment>
        </ThemeProvider>
      </Provider>
    </ClientProvider>
  );
  return { ...renderResult, store: client.store };
}

/*
 * The same as reduxRender() above but also wrapped with 'react-router'
 * in order to test navigation flows.
 *
 * Returns the same object as reduxRender()
 */
export function routerRender(element, initialState, initialRoute = '/') {
  return reduxRender(
    <MemoryRouter initialEntries={[initialRoute]}>{element}</MemoryRouter>,
    initialState
  );
}

/*
 * Declaratively test complex form validations
 *
 * getter   : function that receives a data-testid and returns an HTML element
 * formId   : data-testid of the form element
 * formData : object of type { [data-testid]: any } to populate the form
 * errors   : object of type { [data-testid]: String } with expected errors
 */
export function testValidation(getter, formId, { formData, errors }) {
  // Populate the specified form fields
  Object.keys(formData).forEach((fieldId, i) => {
    const field = getter(fieldId);
    field.value = formData[fieldId];
    Simulate.change(field);
  });

  // Submit the form
  Simulate.submit(getter(formId));

  // Check expected errors
  Object.keys(errors).forEach((fieldId, i) => {
    expect(getter(`${fieldId}-error`).textContent).toEqual(
      expect.stringContaining(errors[fieldId])
    );
  });
}

/*
 * Add dataset property (data-foo attribute values) to nodes
 *
 * This hack is required because jsdom has no support for dataset yet
 * which prevents us from testing components that rely on it.
 * See https://github.com/jsdom/jsdom/issues/961
 *
 * element   : DOM node to add the dataset to
 * dataAttrs : attribute names to add to the node
 *
 * Usage:
 * <button data-testid="btn" data-modal="buy" data-foo="bar">Click</button>
 *
 * const btn = withDataset(getByTestId('btn'), 'modal', 'foo')
 * // btn.dataset === { modal: 'buy', foo: 'bar'}
 *
 */
export function withDataset(element, ...dataAttrs) {
  element.dataset = dataAttrs.reduce((acc, attr) => {
    acc[attr] = element.getAttribute(`data-${attr}`);
    return acc;
  }, {});
  return element;
}

/**
 * Return UTC timestamps relative to the current time
 * Useful for setting up different auction scenarios.
 */
export const twoWeeksAgo = () => Date.now() / 1000 - 60 * 60 * 24 * 7 * 2;
export const oneHourAgo = () => Date.now() / 1000 - 60 * 60;
export const inOneHour = () => Date.now() / 1000 + 60 * 60;
export const inOneWeek = () => Date.now() / 1000 + 60 * 60 * 24 * 7;

/*
 * Returns a common initial state for Redux store that is useful for most tests
 * Accepts an optional object with overrides that will be deeply merged with
 * the base state object.
 */
export function getInitialState(overrides = {}) {
  const baseState = {
    config,
    connectivity: { isOnline: true },
    blockchain: { height: 1, gasPrice: config.DEFAULT_GAS_PRICE },
    converter: {
      status: {
        availableEth: '100',
        availableLmr: '100',
        currentPrice: '10'
      }
    },
    auction: {
      status: {
        nextAuctionStartTime: inOneHour(),
        tokenRemaining: '1',
        currentAuction: 10,
        currentPrice: '33000000000',
        genesisTime: twoWeeksAgo()
      }
    },
    session: { isLoggedIn: true },
    rates: { ETH: { token: 'ETH', price: 1 } },
    wallets: {
      syncStatus: 'up-to-date',
      active: 'foo',
      allIds: ['foo'],
      byId: {
        foo: {
          addresses: {
            '0x15dd2028C976beaA6668E286b496A518F457b5Cf': {
              token: {
                [config.LMR_TOKEN_ADDR]: { balance: '5000000000000000000000' }
              },
              balance: '5000000000000000000000',
              transactions: []
            }
          }
        }
      }
    }
  };
  return merge({}, baseState, overrides);
}

/*
 * A sample transaction object. Useful for populating state in tests.
 */
export function getDummyTransaction() {
  return {
    meta: {
      lumerin: {},
      tokens: {
        [config.LMR_TOKEN_ADDR]: {
          event: 'Transfer',
          processing: false,
          value:
            '0x000000000000000000000000000000000000000000000001e5b8fa8fe2ac0000'
        }
      }
    },
    receipt: {
      blockHash: '0x1234567890abcdef',
      blockNumber: 407574,
      cumulativeGasUsed: 38484,
      gasUsed: 38484,
      transactionHash: '0x1234567890abcdef',
      transactionIndex: 0
    },
    transaction: {
      transactionIndex: 0,
      blockNumber: 407574,
      blockHash: '0x1234567890abcdef',
      hash: '0x1234567890abcdef',
      value: '0',
      gasPrice: '5000000000',
      gas: 76968,
      from: '0x15dd2028C976beaA6668E286b496A518F457b5Cf',
      to: '0xde806D6efD432CDeE42573760682D99eDEdC1d89'
    }
  };
}
