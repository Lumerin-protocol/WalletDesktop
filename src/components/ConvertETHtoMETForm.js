import { sendToMainProcess, isWeiable } from '../utils'
import { DisplayValue, Flex, Btn, Sp } from './common'
import ConfirmationWizard from './ConfirmationWizard'
import ConverterEstimates from './ConverterEstimates'
import MinReturnCheckbox from './MinReturnCheckbox'
import * as validators from '../validator'
import * as selectors from '../selectors'
import AmountFields from './AmountFields'
import { debounce } from 'lodash'
import { connect } from 'react-redux'
import GasEditor from './GasEditor'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import config from '../config'
import React from 'react'
import Web3 from 'web3'

const ConfirmationContainer = styled.div`
  font-size: 1.3rem;
  line-height: 1.8rem;
  font-weight: 600;
  letter-spacing: 0.5px;

  & > div {
    color: ${p => p.theme.colors.primary};
  }
`

const Footer = styled.div`
  background-image: linear-gradient(to bottom, #272727, #323232);
  padding: 3.2rem 2.4rem;
  flex-grow: 1;
  height: 100%;
`

class ConvertETHtoMETForm extends React.Component {
  static propTypes = {
    availableETH: PropTypes.string.isRequired,
    ETHprice: PropTypes.number.isRequired,
    from: PropTypes.string.isRequired,
    tabs: PropTypes.node
  }

  state = {
    ...AmountFields.initialState,
    ...GasEditor.initialState('MET'),
    useMinimum: true,
    estimate: null,
    errors: {},
    rate: null
  }

  onInputChange = e => {
    const { id, value } = e.target
    const { ETHprice } = this.props

    this.setState(state => ({
      ...state,
      ...AmountFields.onInputChange(state, ETHprice, id, value),
      errors: {
        ...state.errors,
        [id]: null,
        useMinimum:
          id === 'estimate' && value !== null ? null : state.errors.useMinimum
      },
      [id]: value
    }))

    // Estimate gas limit again if parameters changed
    if (['ethAmount'].includes(id)) this.getGasEstimate()
  }

  onUseMinimumToggle = () =>
    this.setState(state => ({
      ...state,
      useMinimum: !state.useMinimum,
      errors: {
        ...state.errors,
        useMinimum: null
      }
    }))

  getGasEstimate = debounce(() => {
    const { ethAmount } = this.state

    if (!isWeiable(ethAmount)) return

    sendToMainProcess('metronome-convert-eth-gas-limit', {
      value: Web3.utils.toWei(ethAmount.replace(',', '.')),
      from: this.props.from
    })
      .then(({ gasLimit }) => this.setState({ gasLimit: gasLimit.toString() }))
      .catch(err => console.warn('Gas estimation failed', err)) // eslint-disable-line no-console
  }, 500)

  validate = () => {
    const { ethAmount, gasPrice, gasLimit, estimate, useMinimum } = this.state
    const max = Web3.utils.fromWei(this.props.availableETH)
    const errors = {
      ...validators.validateEthAmount(ethAmount, max),
      ...validators.validateGasPrice(gasPrice, config.MAX_GAS_PRICE),
      ...validators.validateGasLimit(gasLimit),
      ...validators.validateUseMinimum(useMinimum, estimate)
    }
    const hasErrors = Object.keys(errors).length > 0
    if (hasErrors) this.setState({ errors })
    return !hasErrors
  }

  onWizardSubmit = password => {
    return sendToMainProcess('mtn-convert-eth', {
      minReturn:
        this.state.useMinimum && typeof this.state.estimate === 'string'
          ? this.state.estimate
          : undefined,
      gasPrice: Web3.utils.toWei(this.state.gasPrice.replace(',', '.'), 'gwei'),
      gasLimit: this.state.gasLimit,
      password,
      value: Web3.utils.toWei(this.state.ethAmount.replace(',', '.')),
      from: this.props.from
    })
  }

  renderConfirmation = () => {
    const { ethAmount, usdAmount, estimate, rate } = this.state
    return (
      <ConfirmationContainer data-testid="confirmation">
        You will convert{' '}
        <DisplayValue
          inline
          value={Web3.utils.toWei(ethAmount.replace(',', '.'))}
          post=" ETH"
        />{' '}
        (${usdAmount}) and get approximately{' '}
        <DisplayValue value={estimate} post=" MET" inline />
        {', which means a rate of '}
        <DisplayValue inline value={rate} post=" ETH/MET" />.
      </ConfirmationContainer>
    )
  }

  renderForm = goToReview => {
    return (
      <Flex.Column grow="1">
        {this.props.tabs}
        <Sp py={4} px={3}>
          <form
            data-testid="ethToMet-form"
            noValidate
            onSubmit={goToReview}
            id="convertForm"
          >
            <AmountFields
              availableETH={this.props.availableETH}
              ethAmount={this.state.ethAmount}
              usdAmount={this.state.usdAmount}
              autoFocus
              onChange={this.onInputChange}
              errors={this.state.errors}
            />
            <Sp mt={3}>
              <GasEditor
                useCustomGas={this.state.useCustomGas}
                onChange={this.onInputChange}
                gasPrice={this.state.gasPrice}
                gasLimit={this.state.gasLimit}
                errors={this.state.errors}
              />
            </Sp>
            <ConverterEstimates
              convertTo="MET"
              estimate={this.state.estimate}
              onChange={this.onInputChange}
              amount={this.state.ethAmount}
              rate={this.state.rate}
            />
            <MinReturnCheckbox
              useMinimum={this.state.useMinimum}
              onToggle={this.onUseMinimumToggle}
              label="Get expected MET amount or cancel"
              error={this.state.errors.useMinimum}
            />
          </form>
        </Sp>
        <Footer>
          <Btn submit block form="convertForm">
            Review Convert
          </Btn>
        </Footer>
      </Flex.Column>
    )
  }

  render() {
    return (
      <ConfirmationWizard
        renderConfirmation={this.renderConfirmation}
        onWizardSubmit={this.onWizardSubmit}
        pendingTitle="Converting ETH..."
        renderForm={this.renderForm}
        editLabel="Edit this conversion"
        validate={this.validate}
      />
    )
  }
}

const mapStateToProps = state => ({
  availableETH: selectors.getEthBalanceWei(state),
  ETHprice: selectors.getEthRate(state),
  from: selectors.getActiveWalletAddresses(state)[0]
})

export default connect(mapStateToProps)(ConvertETHtoMETForm)
