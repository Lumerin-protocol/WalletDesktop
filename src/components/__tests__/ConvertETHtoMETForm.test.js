import ConvertETHtoMETForm from '../ConvertETHtoMETForm'
import * as amountFields from './AmountFields.test.js'
import * as gasEditor from './GasEditor.test.js'
import * as testUtils from '../../testUtils'
import { Simulate } from 'react-testing-library'
import config from '../../config'
import React from 'react'

const element = <ConvertETHtoMETForm />

const ETHprice = 250

describe('<ConvertETHtoMETForm/>', () => {
  it('should match its snapshot', () => {
    const { container } = testUtils.reduxRender(element, getInitialState())
    expect(container).toMatchSnapshot()
  })

  describe('When editing the amount fields', () => {
    amountFields.runEditTests(element, getInitialState(), ETHprice)
  })

  describe('When submitting the form', () => {
    amountFields.runValidateTests(element, getInitialState(), 'ethToMet-form')

    gasEditor.runValidateTests(element, getInitialState(), 'ethToMet-form')

    it('displays the confirmation view if there are no errors', () => {
      const { queryByTestId, getByTestId } = testUtils.reduxRender(
        element,
        getInitialState()
      )
      expect(queryByTestId('confirmation')).toBeNull()
      const amountField = getByTestId('ethAmount-field')
      amountField.value = '1'
      Simulate.change(amountField)
      Simulate.submit(getByTestId('ethToMet-form'))
      expect(queryByTestId('confirmation')).not.toBeNull()
    })
  })
})

function getInitialState() {
  return {
    connectivity: { isOnline: true },
    blockchain: { height: 1 },
    metronome: { transferAllowed: true },
    converter: {
      status: {
        currentPrice: '10'
      }
    },
    auction: { status: null },
    session: { isLoggedIn: true },
    rates: { ETH: { token: 'ETH', price: ETHprice } },
    wallets: {
      active: 'foo',
      allIds: ['foo'],
      byId: {
        foo: {
          addresses: {
            '0xf00': {
              token: { [config.MTN_TOKEN_ADDR]: { balance: '0' } },
              balance: '5000000000000000000000'
            }
          }
        }
      }
    }
  }
}
