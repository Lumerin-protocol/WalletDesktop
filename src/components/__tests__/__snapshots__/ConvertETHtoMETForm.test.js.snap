// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AmountFields/> should match its snapshot 1`] = `
<div>
  <div
    class="sc-jKJlTe fkNCah"
  >
    <div
      class="sc-ckVGcZ kvHIve"
    >
      <button
        type="button"
        class="sc-htpNat fdEoJm"
        data-testid="max-btn"
        tabindex="-1"
      >
        MAX
      </button>
      <div>
        <label
          class="sc-dnqmqq ilsafR"
          for="ethAmount"
        >
          Amount (ETH)
        </label>
        <input
          type="text"
          class="sc-iwsKbI kZiHEX"
          value=""
          id="ethAmount"
          data-testid="ethAmount-field"
          placeholder="0.00"
        />
      </div>
    </div>
    <div
      class="sc-hMqMXs dnTKXF"
    >
      <svg
        class="sc-bxivhb hXQRLM"
        viewBox="0 0 24 24"
      >
        <path
          d="M24 7.136v.96H1.642l4.464 4.464-.682.682-5.28-5.28a.475.475 0 0 1 0-.682L5.424 2l.682.682-4.464 4.464H24v-.01zM18.576 11.6l-.682.682 4.464 4.464H0v.96h22.358l-4.464 4.464.682.681 5.28-5.28a.475.475 0 0 0 0-.681l-5.28-5.29z"
        />
      </svg>
    </div>
    <div
      class="sc-ckVGcZ kvHIve"
    >
      <div>
        <label
          class="sc-dnqmqq ilsafR"
          for="usdAmount"
        >
          Amount (USD)
        </label>
        <input
          type="text"
          class="sc-iwsKbI kZiHEX"
          value=""
          id="usdAmount"
          data-testid="usdAmount-field"
          placeholder="0.00"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<ConvertETHtoMETForm/> should match its snapshot 1`] = `
<div>
  <div
    class="sc-eNQAEJ hdurla"
  >
    <div
      class="sc-hMqMXs dKMJsu"
    >
      <form
        data-testid="ethToMet-form"
        novalidate=""
        id="convertForm"
      >
        <div
          class="sc-jKJlTe fkNCah"
        >
          <div
            class="sc-ckVGcZ kvHIve"
          >
            <button
              type="button"
              class="sc-htpNat fdEoJm"
              data-testid="max-btn"
              tabindex="-1"
            >
              MAX
            </button>
            <div>
              <label
                class="sc-dnqmqq ilsafR"
                for="ethAmount"
              >
                Amount (ETH)
              </label>
              <input
                type="text"
                class="sc-iwsKbI kZiHEX"
                value=""
                id="ethAmount"
                data-testid="ethAmount-field"
                placeholder="0.00"
              />
            </div>
          </div>
          <div
            class="sc-hMqMXs dnTKXF"
          >
            <svg
              class="sc-bxivhb hXQRLM"
              viewBox="0 0 24 24"
            >
              <path
                d="M24 7.136v.96H1.642l4.464 4.464-.682.682-5.28-5.28a.475.475 0 0 1 0-.682L5.424 2l.682.682-4.464 4.464H24v-.01zM18.576 11.6l-.682.682 4.464 4.464H0v.96h22.358l-4.464 4.464.682.681 5.28-5.28a.475.475 0 0 0 0-.681l-5.28-5.29z"
              />
            </svg>
          </div>
          <div
            class="sc-ckVGcZ kvHIve"
          >
            <div>
              <label
                class="sc-dnqmqq ilsafR"
                for="usdAmount"
              >
                Amount (USD)
              </label>
              <input
                type="text"
                class="sc-iwsKbI kZiHEX"
                value=""
                id="usdAmount"
                data-testid="usdAmount-field"
                placeholder="0.00"
              />
            </div>
          </div>
        </div>
        <div
          class="sc-hMqMXs bQjXee"
        >
          <div
            class="sc-jKJlTe kWLJKD"
          >
            <div
              class="sc-jKJlTe jGctBw"
            >
              <span
                class="sc-jAaTju fRTHaR"
              >
                Gas Limit: 
                2000000
                (Units)
              </span>
              <span
                class="sc-jAaTju fRTHaR"
              >
                Gas Price: 
                1
                (Gwei)
              </span>
            </div>
            <button
              type="button"
              class="sc-htpNat henWpP"
              data-testid="edit-gas-btn"
              tabindex="-1"
            >
              EDIT GAS
            </button>
          </div>
        </div>
      </form>
    </div>
    <div
      class="sc-gPEVay iUpDMu"
    >
      <button
        type="submit"
        class="sc-bwzfXH emHLVs"
        form="convertForm"
      >
        Review Convert
      </button>
    </div>
  </div>
</div>
`;

exports[`<GasEditor/> should match its snapshot 1`] = `
<div>
  <div
    class="sc-jKJlTe fkNCah"
  >
    <div
      class="sc-ckVGcZ kvHIve"
    >
      <div>
        <label
          class="sc-dnqmqq ilsafR"
          for="gasLimit"
        >
          Gas Limit (Units)
        </label>
        <input
          type="text"
          class="sc-iwsKbI kZiHEX"
          value="21000"
          id="gasLimit"
          data-testid="gas-limit-field"
        />
      </div>
    </div>
    <div
      class="sc-hMqMXs keWZLk"
    />
    <div
      class="sc-ckVGcZ kvHIve"
    >
      <div>
        <label
          class="sc-dnqmqq ilsafR"
          for="gasPrice"
        >
          Gas Price (Gwei)
        </label>
        <input
          type="text"
          class="sc-iwsKbI kZiHEX"
          value="1"
          id="gasPrice"
          data-testid="gas-price-field"
        />
      </div>
    </div>
  </div>
</div>
`;
